### DISCUSSION TITLE
Pass parametrs for each record in custom list records page

### DISCUSSION THREAD

**Discussion by dmitrijsmihailovs** [filamentphp/filament#7816] [Help]
I am trying to create tree view page and for each record I need "Add child" action that redirects to create page with additional parameter `/create?parentId=x` and then set that parameter maybe in hidden input?

I am trying to trigger that create action like this
`wire:click="mountAction('create', { 'parentId': {{ $model->getKey() }} })"`
But it does not work.

**Comment by pboivin** [comment]
If this is simply redirecting to a new page, can you use a url instead for your action? Example here:
https://filamentphp.com/docs/3.x/actions/trigger-button#setting-a-label

**Comment by micraux** [comment]
Try this:
```php
Action::make('add_child')
    ->url(
        YourResource::getUrl(
            'create',
            ['parentId' => your_value]
        )
    )
```
